src/content/docs/en/guides/content-collections.mdx

```diff
@@ -543,6 +543,48 @@ If you have an existing Astro project, such as a blog, that uses Markdown or MDX
  
 See how to convert a basic blog example from `src/pages/posts/` to `src/content/posts` in our [step-by-step tutorial](/en/tutorials/add-content-collections/) that uses the codebase from [the Build a Blog tutorial's finished project](https://github.com/withastro/blog-tutorial-demo).
 
+## Enabling JSON Schema Generation
+
+<p><Since v="4.12.0" /></p>
+
+If you are working with data collections, you may wish to generate JSON Schema files to get better IntelliSense and type-checking in your editor. Astro provides a flag to auto-generate a JSON Schema for content collections of `type: 'data'` which can be used as the `$schema` value. This feature generates JSON Schema files for each data collection in your project, using a library called [zod-to-json-schema](https://github.com/StefanTerdell/zod-to-json-schema#known-issues)
+
+To enable this feature, add the flag:
+
+```astro.config.mjs
+import { defineConfig } from 'astro/config';
+
+export default defineConfig({
+	contentCollection: {
++		JsonSchema: true
+	}
+});
+```
+
+This feature requires you to manually reference the schema, which you can either do in each file or in your editor settings:
+
+```src/content/test/entry.json
```

Comment:
``````suggestion
```json title="src/content/authors/armand.json" ins={2}
``````
Now that this is not experimental, can we have a more realistic example here? And format it using Expressive code to highlight the important line?

Reply:
I'll take a look at the example tomorrow morning _(CET)_